src/hotspot/share/runtime/signature.cpp

```diff
@@ -188,7 +188,7 @@ void Fingerprinter::compute_fingerprint_and_return_type(bool static_flag) {
 #endif
 
   // Detect overflow.  (We counted _param_size correctly.)
-  if (_method == nullptr && _param_size > fp_max_size_of_parameters) {
+  if (_method == nullptr && _param_count > fp_max_size_of_parameters) {
```

Comment:
Seems like `fp_max_size_of_parameters` ought to be renamed.  But I found other uses
that are comparing to something else, so maybe there are problems elsewhere.
Also, the use of "size" in both the preceding and following comments needs to be updated.